src/OrchardCore/OrchardCore.DisplayManagement/Zones/FlatPositionComparer.cs

```diff
@@ -68,34 +94,110 @@ public int Compare(string x, string y)
             {
                 return -1;
             }
+
+            partIndex++;
         }
 
-        // All things being equal y might be further defined than x (e.g. x == 1.2 and y == 1.2.3)
-        if (xParts.Length < yParts.Length)
+        // Check if y has more parts
+        if (yParts.MoveNext())
         {
             return -1;
         }
 
         return 0;
     }
 
-    private static string NormalizeKnownPartitions(string partition)
+    private static ReadOnlySpan<char> GetNormalizedSpan(string value)
+    {
+        if (value is null)
+        {
+            return "before.".AsSpan();
+        }
+
+        var span = value.AsSpan();
+        if (span.IsWhiteSpace())
+        {
+            return "0".AsSpan();
+        }
+
+        // Trim ':' from start and '.' from end
+        span = span.Trim(':');
+        span = span.TrimEnd('.');
+
+        return span;
+    }
+
+    private static bool TryNormalizeKnownPartitions(ReadOnlySpan<char> partition, out int position)
     {
         if (partition.Length < 5) // known partitions are long
         {
-            return partition;
+            position = 0;
+            return false;
         }
 
-        if (string.Equals(partition, "before", StringComparison.OrdinalIgnoreCase))
+        if (partition.Equals("before".AsSpan(), StringComparison.OrdinalIgnoreCase))
```

Comment:
I am surprised you need to call `AsSpan()` on all these strings. There should be a default conversion, or maybe it's with recent SDKs only.

Reply:
I don't think this is required, it's just more clear. I'll remove this.

Reply:
I've removed all `AsSpan` occurrences.